extensions/danbooru/src/components/danbooruList.tsx

```diff
@@ -0,0 +1,77 @@
+import { showToast, Grid, getPreferenceValues } from "@raycast/api";
+import { useState, useEffect } from "react";
+import fetch from "node-fetch";
+import { DanbooruListProps, PostDetailsProps, PostProps } from "../types/types";
+import { ListActions } from "./listActions";
+
+export default function DanbooruList(props: DanbooruListProps) {
+  const [posts, setPosts] = useState<PostProps[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  async function fetchPosts() {
+    const preferences = getPreferenceValues();
+    const login = preferences.username;
+    const apiKey = preferences.apiKey;
+    const postsUrl = props.sfw
+      ? `https://danbooru.donmai.us/posts.json?login=${login}&api_key=${apiKey}&tags=${props.tag1}+${props.tag2}+rating%3Ag&limit=${props.numberOfPosts}`
+      : `https://danbooru.donmai.us/posts.json?login=${login}&api_key=${apiKey}&tags=${props.tag1}+${props.tag2}&limit=${props.numberOfPosts}`;
+
+    try {
+      const response = await fetch(postsUrl, {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+
+      const posts: PostProps[] = (await response.json()) as PostProps[];
+      setPosts(posts);
+    } catch (error) {
+      if (error instanceof Error) {
+        showToast({ title: "Error", message: error.message + ". Did you correctly set up your API Key ?" });
+      } else {
+        showToast({ title: "Error", message: "An unknown error occurred" });
```

Comment:
Could we use [failure style](https://developers.raycast.com/api-reference/feedback/toast#toast.style) when something goes wrong?

Reply:
added in 58d8a571c3f8a59ef93e56544237d6c4d1484cfe. thanks for pointing it out!